File: bill-api/Domain/Entities/Bill.cs
using bill_api.Domain.Common;

namespace bill_api.Domain.Entities
{
    public class Bill : BaseEntity
    {

        public required string BillCode { get; set; }

        public double? BillTotal { get; set; }

        public double? Subtotal { get; set; }

        public double? VAT { get; set; }

        public double? Withholding { get; set; }

        public DateTime CreationDate { get; set; }

        public string? Status { get; set; }

        public string? City { get; set; }

        public bool? IsPaid { get; set; }

        public DateTime? PaymentDate { get; set; }

        public string? ClientId { get; set; }

    }
}

File: bill_app/lib/domain/entities/bill.dart
import 'dart:convert';

class Bill {
  String billCode;
  double? billTotal;
  double? subtotal;
  double? vat;
  double? withholding;
  DateTime creationDate;
  String? status;
  String? city;
  bool? isPaid;
  DateTime? paymentDate;
  String? clientId;

  Bill({
    required this.billCode,
    this.billTotal,
    this.subtotal,
    this.vat,
    this.withholding,
    required this.creationDate,
    this.status,
    this.city,
    this.isPaid,
    this.paymentDate,
    this.clientId,
  });

  factory Bill.fromJson(String json) => Bill.fromMap(jsonDecode(json));

  String toJson() => jsonEncode(toMap());

  factory Bill.fromMap(Map<String, dynamic> map) {
    return Bill(
      billCode: map['billCode'],
      billTotal: double.parse(map['billTotal'].toString()),
      subtotal: double.parse(map['subtotal'].toString()),
      vat: double.parse(map['vat'].toString()),
      withholding: double.parse(map['withholding'].toString()),
      creationDate: DateTime.parse(map['creationDate']),
      status: map['status'],
      city: map['city'],
      isPaid: map['isPaid'],
      paymentDate: map['paymentDate'] != null
          ? DateTime.parse(map['paymentDate'])
          : null,
      clientId: map['clientId'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'billCode': billCode,
      'billTotal': billTotal,
      'subtotal': subtotal,
      'vat': vat,
      'withholding': withholding,
      'creationDate': creationDate.toIso8601String(),
      'status': status,
      'city': city,
      'isPaid': isPaid,
      'paymentDate': paymentDate?.toIso8601String(),
      'clientId': clientId,
    };
  }
}
